### PR TITLE
Visualise `source_location`

### PR DESCRIPTION
**Pull request by SuperWig**
Maybe the function name could be included in the display string? Thought that could get a be a bit long though so opted for just the line and column number. Also not sure if I did the strings correctly 🤷‍♂️

### CODE REVIEW

File: stl/debugger/STL.natvis [hunk 1]
```diff
@@ -1827,4 +1827,14 @@ SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
     </Expand>
   </Type>
 
+  <Type Name="std::source_location">
+    <DisplayString>line: {_Line}, column: {_Column}</DisplayString>
```
Comment (SuperWig) [inline stl/debugger/STL.natvis:1831]: Perhaps?
```suggestion
    <DisplayString>{_Function,sb}: ({_Line}, {_Column})</DisplayString>
```

Comment (AnjuDel) [reply stl/debugger/STL.natvis:1831]: I agree that I prefer the second option more

**Review by AnjuDel — CHANGES_REQUESTED** [review]

**Review by AnjuDel — APPROVED** [review]

**Review by StephanTLavavej — APPROVED** [review]

**Comment by StephanTLavavej** [comment]
Looks good to me. @AnjuDel, there will be a merge conflict (adjacent adds) with #1855. You can decide how you want to handle this:

1. Separate mirrors:
   1. Merge GH-1855, MSVC-PR-319334, and VS-PR-319336.
   2. Resolve the merge conflict in this PR.
   3. Port this PR to MSVC and VS, then (after they're approved and green) merge simultaneously.
2. Consolidated mirrors:
   1. Update MSVC-PR-319334 and VS-PR-319336 to add this PR's content (choosing where to add `source_location`).
   2. Merge GH-1855, MSVC-PR-319334, and VS-PR-319336.
   3. Resolve the merge conflict in this PR. (Important: This should produce an identical file to what was just merged in MSVC and VS, without whitespace or ordering differences.)
   4. Merge this PR.
   
I am fine with either process as long as no codebase divergence is introduced.

**Review by StephanTLavavej — CHANGES_REQUESTED** [review]
The checks are failing because:

```
Validation failed: .\stl\debugger\STL.natvis contains LF line endings.
Validation failed: .\stl\debugger\STL.natvis contains 1 lines with trailing whitespace.
```

File: stl/debugger/STL.natvis [hunk 2]
```diff
@@ -1827,6 +1827,15 @@ SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
     </Expand>
   </Type>
 
+  <Type Name="std::source_location">
```
Comment (StephanTLavavej) [inline stl/debugger/STL.natvis:1830]: `</Type>` is missing.

**Review by StephanTLavavej — CHANGES_REQUESTED** [review]

**Review by StephanTLavavej — APPROVED** [review]

**Comment by StephanTLavavej** [comment]
Approved the internal mirrors MSVC-PR-323231 and VS-PR-323226, thanks @AnjuDel!

**Pull request merged**